### PR TITLE
Add Zero Inflated Distributions

### PR DESCRIPTION
**Pull request by juanitorduz**
Add Zero Inflated Distributions + tests

### CODE REVIEW

File: distributions/zi_binomial.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    return ppf(1.0 - q, psi, n, p)
+
+
+def rvs(psi, n, p, size=None, random_state=None):
+    base_samples = pt.random.binomial(n, p, size=size, rng=random_state)
+    mask = pt.random.bernoulli(psi, size=size)
+    return pt.cast(mask, "int64") * base_samples
+
+
+def logcdf(x, psi, n, p):
```
Comment (aloctavodia) [inline distributions/zi_binomial.py]: Better if we can implement this directly

File: distributions/zi_binomial.py [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+def std(psi, n, p):
+    return pt.sqrt(var(psi, n, p))
+
+
+def skewness(psi, n, p):
+    q = 1 - p
+    base_ex1 = n * p
+    base_ex2 = n * p * q + base_ex1 * base_ex1
+    base_ex3 = n * p * q * (1 - 2 * p) + 3 * n * n * p * p * q + base_ex1 * base_ex1 * base_ex1
```
Comment (aloctavodia) [inline distributions/zi_binomial.py]: A few products could be powers, here and in other places

Comment (juanitorduz) [reply distributions/zi_binomial.py]: yes, fixed in https://github.com/pymc-devs/distributions/pull/18/changes/7f9d849f7a37881a6a10b9a4d1855eefa55ef590

File: distributions/zi_binomial.py (hunk 1 above)
Comment (juanitorduz) [reply distributions/zi_binomial.py]: fixed in https://github.com/pymc-devs/distributions/pull/18/changes/7f9d849f7a37881a6a10b9a4d1855eefa55ef590

File: tests/test_zi_binomial.py [hunk 3]
```diff
@@ -0,0 +1,193 @@
+"""Test Zero-Inflated Binomial distribution using empirical validation."""
```
[the remark this answers is not in the document]
Comment (juanitorduz) [reply tests/test_zi_binomial.py]: Tried it but could not find an immediate way:

The `run_empirical_tests` helper has strict PPF edge-case tests that expect specific behavior for out-of-bound quantile values, which doesn't work correctly for discrete distributions with finite upper bounds (like zi_binomial's `[0, n]` support). For zi_poisson with unbounded support `[0, ∞)`, the helper works fine, so it continues to use `run_empirical_tests`.

File: distributions/zi_binomial.py [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+from distributions.helper import cdf_bounds, discrete_entropy
+from distributions.optimization import find_ppf_discrete
+
+
+def mean(psi, n, p):
+    return psi * n * p
+
+
+def mode(psi, n, p):
+    return pt.zeros_like(psi * n * p)
```
Comment (aloctavodia) [inline distributions/zi_binomial.py]: In Preliz we compute the pmf over a reasonable range and then pick the argmax, but we could do this instead

```python
def mode(psi, n, p):
    base_mode = binomial_mode(n, p)
    return pt.switch(logpdf(0, psi, n, p) >=  logpdf(base_mode, psi, n, p), 0, base_mode)
```
The same applies to the other ZI distributions, so we could have a general function in `helper.py` that we reuse.

File: tests/test_zi_binomial.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (aloctavodia) [reply tests/test_zi_binomial.py]: Yes, we have that to match scipy's behaviour. Weird because `find_ppf_discrete` should handle that. Let's keep this for the moment I will then check if we need to improve something else.
